Newspack/NewspackShare/Controllers/ShareMediaViewController.swift

```diff
@@ -0,0 +1,331 @@
+    }
+
+    func configureShadows() {
+        shadowSites = shadowManager.retrieveShadowSites()
+
+        guard
+            let currentSiteIdentifer = UserDefaults.shared.string(forKey: AppConstants.currentSiteIDKey),
+            let currentStoryIdentifier = UserDefaults.shared.string(forKey: AppConstants.lastSelectedStoryFolderKey + currentSiteIdentifer)
+        else {
+            return
+        }
+        for site in shadowSites {
+            for story in site.stories {
+                if story.uuid == currentStoryIdentifier {
+                    targetStory = story
+                    return
+                }
+            }
+        }
+    }
+
+    func configureSharedItems() {
+        extracter.loadShare { (extracted) in
+            self.handleSharedItems(items: extracted)
+        }
+    }
+
+}
+
+// MARK: - Validation and prompts
+
+extension ShareMediaViewController {
+
+    /// In order to share to the main app, the main app must have at least one
+    /// site and at least one story. Since we don't have direct access to the main
+    /// app's data, we'll trust that shadowSites are an accurate reflection. If
+    /// either shadowSites are empty, or there is not a target story retrieved
+    /// we'll assume the app has no sites or stories.
+    /// Note that there is an edge case where shadowSites and a targetStory could
+    /// be stale, and the app actually have no sites or stories (e.g. if the site
+    /// or story folders were manually deleted via the files app.
+    ///
+    func validateData() -> Bool {
+        guard
+            shadowSites.count > 0,
+            let _ = targetStory
+        else {
+            return false
+        }
+
+        return true
+    }
+
+    func interruptSharing() {
+        stackView.isHidden = true
+
+        let alertTitle = NSLocalizedString("Unable to share", comment: "The title of an error message shown when a user tries to share but the app is not in a state where sharing is possible..")
+        let alertMessage = NSLocalizedString("Please launch the Newspack app, log in to your site, and make sure you have at least one story to share to, then try again.", comment: "A message shown in a prompt when trying to share to the app but there are not yet any sites or stories in the app.")
+        let actionTitle = NSLocalizedString("OK", comment: "OK. A button title. Tapping closes the share screen.")
+        let alert = UIAlertController(title: alertTitle, message: alertMessage, preferredStyle: .alert)
+
+        let action = UIAlertAction(title: actionTitle, style: .default, handler: { [weak self] _ in
+            self?.cancelSharing()
+        })
+        alert.addAction(action)
+
+        present(alert, animated: true, completion: nil)
+    }
+
+    func cancelSharing() {
+        let error = NSError(domain: "com.auttomattic.newspack.share", code: 0, userInfo: nil)
```

Comment:
Would the domain be worthy of a constant?

Reply:
Update: Probably not!!

Reply:
No no... good call actually.  As more types are supported by the share extension there may be more opportunities for dispatching/logging NSErrors.